### PR TITLE
Add Rubocop support

### PR DESCRIPTION
**Pull request by ErikSchierboom**

### CODE REVIEW

File: .github/config/rubocop_linter_action.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+check_name: "Rubocop Results"
```
Comment (iHiD) [inline .github/config/rubocop_linter_action.yml:1]: I think this file is unnecessary.

File: bin/rubocop-quick [hunk 2]
```diff
@@ -0,0 +1,4 @@
+#!/bin/bash
+
+git diff --name-status --staged | grep '^[MA]' | grep -o '\s\+.*rb' | xargs bundle exec rubocop --except Metrics --auto-correct --format quiet && \
```
Comment (iHiD) [inline bin/rubocop-quick:3]: You need a file name at the end of this before the `&&`. Probably `Gemfile` is the right one.

File: lib/minitest_ext/exercism_plugin.rb [hunk 3]
```diff
@@ -1,8 +1,7 @@
 module Minitest
-  def self.plugin_bogus_options(opts, options)
-  end
+  def self.plugin_bogus_options(opts, options); end
 
-  def self.plugin_exercism_init(*args)
+  def self.plugin_exercism_init(*_args)
```
Comment (iHiD) [inline lib/minitest_ext/exercism_plugin.rb:4]: Why this change? Is this Rubocop?

File: lib/minitest_ext/exercism_reporter.rb [hunk 4]
```diff
@@ -30,16 +25,16 @@ def report
     end
 
     def status
-      tests.all?{|t|t[:status] == :pass} ? :pass : :fail
+      tests.all? { |t| t[:status] == :pass } ? :pass : :fail
     end
 
-    def exception_raised!(e)
+    def exception_raised!(exc)
```
Comment (iHiD) [inline lib/minitest_ext/exercism_reporter.rb:31]: Is this Rubocop too?

Comment (ErikSchierboom) [reply lib/minitest_ext/exercism_reporter.rb:31]: It is. This was the rule about an identifier having to be at least three characters long. Is that okay, or do you want me to ignore it?

Comment (iHiD) [reply lib/minitest_ext/exercism_reporter.rb:31]: Yeah - ignore that rule please. 

We should somewhere keep a global rubocop file with all these in that can act as our base, as we're discovering extra useful things in every repo but they're all getting out of sync.

File: .github/config/rubocop_linter_action.yml (hunk 1 above)
Comment (ErikSchierboom) [reply .github/config/rubocop_linter_action.yml:1]: I copied this from the v3-website, which had this file. Shall I remove and see if it still works?

File: lib/minitest_ext/exercism_plugin.rb (hunk 3 above)
Comment (ErikSchierboom) [reply lib/minitest_ext/exercism_plugin.rb:4]: Yes:

```bash
== lib/minitest_ext/exercism_plugin.rb ==
W:  4: 34: [Corrected] Lint/UnusedMethodArgument: Unused method argument - args. If it's necessary, use _ or _args as an argument name to indicate that it won't be used. You can also write as plugin_exercism_init(*) if you want the method to accept any arguments but don't care about them.
```

File: lib/minitest_ext/exercism_reporter.rb (hunk 4 above)
Comment (kotp) [reply lib/minitest_ext/exercism_reporter.rb:31]: Yes, please do use the common names for the pretty much industry wide standards.  `i` for index rather than `ndx` or something else.  `e` for `exception` as it is a pronoun that is clearly defined by proximity, etc.

File: lib/minitest_ext/exercism_plugin.rb (hunk 3 above)
Comment (kotp) [reply lib/minitest_ext/exercism_plugin.rb:4]: Historically, I think, prefer `*_` as opposed to `*_args` because context defines that they are "arguments".  And we avoid the poor name of "args".

File: .github/config/rubocop_linter_action.yml (hunk 1 above)
Comment (iHiD) [reply .github/config/rubocop_linter_action.yml:1]: Please! And if so, remove it from there too :) I think this was only needed for a previous iteration of this, but I've since rewritten it.

File: lib/minitest_ext/exercism_plugin.rb (hunk 3 above)
Comment (iHiD) [reply lib/minitest_ext/exercism_plugin.rb:4]: > You can also write as `plugin_exercism_init(*)`

Let's do this. I didn't know it was possible but I like it.

Comment (kotp) [reply lib/minitest_ext/exercism_plugin.rb:4]: The `*` means that you can't use what is given, while the `*_` gives you at least something to work with if you need to, and is idiomatic, though it does communicate "something we are accept but don't use" in block and method argument lists.

Comment (iHiD) [reply lib/minitest_ext/exercism_plugin.rb:4]: > while the *_ gives you at least something to work with if you need to

I think at that stage, the method should just be renamed to `*args` though. I prefer enforcing that we *can't* use them, if the intention is not to use them.
